src/test/kotlin/org/koil/image/PersistedImageServiceTest.kt

```diff
@@ -0,0 +1,70 @@
+package org.koil.image
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import org.junit.jupiter.api.Test
+import org.koil.BaseIntegrationTest
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.mock.web.MockMultipartFile
+
+class PersistedProfileImageServiceTest : BaseIntegrationTest() {
+
+    @Autowired
+    lateinit var profileImageService: ProfileImageService
+
+    val example = MockMultipartFile(
+        "large.jpg",
+        "large.jpg",
+        "IMAGE/JPEG",
+        this::class.java.getResourceAsStream("/img/large.jpg")!!
+    )
+    val alternative = MockMultipartFile(
+        "large.png",
+        "large.png",
+        "IMAGE/PNG",
+        this::class.java.getResourceAsStream("/img/large.png")!!
+    )
+
+    @Test
+    internal fun `GIVEN a user hasn't set their image yet WHEN setting image THEN do so successfully`() {
+        withTestAccount { user ->
+            val result = profileImageService.saveProfileImage(user.accountId!!, example)
+
+            assertThat(result.id).isNotNull()
+
+            val retrieved = profileImageService.getProfileImageUrl(user.accountId!!)
+
+            assertThat(retrieved)
+                .isEqualTo(result.publicId.toString())
+        }
+    }
+
+    @Test
+    internal fun `GIVEN a user has set their image WHEN setting image again THEN overwrite the image and record updated`() {
```

Comment:
I haven't read through the logic yet- but I'm wondering what we do with the old image. Is it still in storage or do we override it with the new profile image?